astropy/time/core.py

```diff
@@ -1787,6 +1787,85 @@ def light_travel_time(self, skycoord, kind='barycentric', location=None, ephemer
         tcor_val = (spos * cpos).sum(axis=-1) / const.c
         return TimeDelta(tcor_val, scale='tdb')
 
+    def to_local(self, theta, longitude):
+        """
+        Adds longitude including the TIO locator rigorously corrected
+        for polar motion.
+
+        Parameters
+        ---------------
+        theta : float or float array
+            Earth rotation angle or Greenwich sidereal time in radians.
+        longitude : `~astropy.coordinates.Longitude`
+            The longitude on the Earth at which to compute the result.
+
+        Returns
+        -------
+        local_angle : float or float array
+            Either local Earth rotation angle or local sidereal time depending
+            on theta.
+        """
+
+        from astropy.coordinates.builtin_frames.utils import get_polar_motion
+        from astropy.coordinates.matrix_utilities import rotation_matrix
+
+        jd1_tt = getattr(self, 'tt')._time.jd1
+        jd2_tt = getattr(self, 'tt')._time.jd2_filled
+        sp = erfa.sp00(jd1_tt, jd2_tt)
+        xp, yp = get_polar_motion(self._time)
+        lon = longitude.lon.to_value(u.radian)
```

Comment:
This is not necessary; also, I think here `longitude` is an actual longitude, not an `location`, so this should really error. That it doesn't suggests we are not testing this path...